packages/loot-core/src/server/schedules/app.js

```diff
@@ -176,6 +176,15 @@ export async function setNextDate({ id, start, conditions, reset }) {
 
 // Methods
 
+export async function checkIfScheduleExists(name, scheduleId) {
+  let idForName = await db.first('SELECT id from schedules WHERE name = ?', [
+    name,
+  ]);
+  return (
+    idForName !== null && (scheduleId ? idForName['id'] !== scheduleId : true)
+  );
```

Comment:
Tried to make this a little easier to follow, let me know what you think:

```suggestion
  let idForName = await db.first('SELECT id from schedules WHERE name = ?', [
    name,
  ]);

  if (idForName == null) {
    return false;
  }
  if (scheduleId) {
    return idForName['id'] !== scheduleId;
  }
  return true;
```